packages/react-native/ReactCommon/react/renderer/textlayoutmanager/platform/ios/react/renderer/textlayoutmanager/RCTTextLayoutManager.mm

```diff
@@ -152,15 +152,17 @@ - (LinesMeasurements)getLinesForAttributedString:(facebook::react::AttributedStr
                                                      attribute:NSFontAttributeName
                                                        atIndex:0
                                                 effectiveRange:nil];
+                                            CGRect lineRect = [layoutManager lineFragmentRectForGlyphAtIndex:range.location effectiveRange:nil];
```

Comment:
Why are we getting line rect for first glyph of the line, instead of using the rect for the line fragment?

(I think the "fragment" part is because layout allows holes in the middle of the lines. https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/TextLayout/Concepts/CalcTextLayout.html)

Reply:
I think I thought there were some differences between the two but now I cannot find anything related to it in the docs, so I may have confused it with something else.

I've updated the code to use the fragment line rect in https://github.com/facebook/react-native/pull/45102/commits/828deaa1973672a5f705d724df859b9dbb19bcdf